_studio/mfx_lib/encode_hw/hevc/agnostic/g9/hevcehw_g9_legacy_defaults.cpp

```diff
@@ -2570,23 +2570,31 @@ class CheckAndFix
 
     static mfxStatus WinBRC(
         Defaults::TCheckAndFix::TExt
-        , const Defaults::Param&
+        , const Defaults::Param& dpar
         , mfxVideoParam& par)
     {
         mfxExtCodingOption3* pCO3 = ExtBuffer::Get(par);
-        MFX_CHECK(pCO3, MFX_ERR_NONE);
+        MFX_CHECK(pCO3 && (pCO3->WinBRCSize || pCO3->WinBRCMaxAvgKbps), MFX_ERR_NONE);
 
-        mfxU32 changed = 0;
-        bool   bSWBRC  = Legacy::IsSWBRC(par, ExtBuffer::Get(par));
-        
-        changed +=
-            (par.mfx.RateControlMethod != MFX_RATECONTROL_VBR || !bSWBRC)
-            && (   CheckOrZero<mfxU16, 0>(pCO3->WinBRCSize)
-                || CheckOrZero<mfxU16, 0>(pCO3->WinBRCMaxAvgKbps));
-
-        changed +=
-            (pCO3->WinBRCSize > 0 || pCO3->WinBRCMaxAvgKbps > 0)
-            && CheckMinOrClip(pCO3->WinBRCMaxAvgKbps, par.mfx.TargetKbps);
+        mfxExtCodingOption2* pCO2 = ExtBuffer::Get(par);
+        bool   bExtBRC      = pCO2 && IsOn(pCO2->ExtBRC);
+        bool   bVBR         = (par.mfx.RateControlMethod == MFX_RATECONTROL_VBR || par.mfx.RateControlMethod == MFX_RATECONTROL_QVBR);
+        bool   bUnsupported = bVBR && bExtBRC && pCO3->WinBRCMaxAvgKbps && (pCO3->WinBRCMaxAvgKbps < TargetKbps(par.mfx));
+        mfxU32 changed      = !bVBR || bUnsupported;
```

Comment:
I think 'changed' flag should correspond to changes in input parameters followed by corresponding warning on Query()

Reply:
It is OK,  'changed' resets pCO3->WinBRCSize and pCO3->WinBRCMaxAvgKbps to zero